config/dev.exs

```diff
@@ -40,3 +40,10 @@ config :phoenix, :stacktrace_depth, 20
 
 # Initialize plugs at runtime for faster development compilation
 config :phoenix, :plug_init_mode, :runtime
+
+config :k8s,
```

Comment:
This is definitely not what you want to do. When having an in-cluster config, you want to use a ServiceAccount. I think this is explained here with the env vars: https://github.com/coryodaniel/k8s/blob/master/lib/k8s/conn/config.ex#L9

In general, by default you should always care about the serviceAccount, and optionally allow other methods, which are probably useful only when deploying out of the cluster